docs/guides/chip_tool_guide.md

```diff
@@ -0,0 +1,807 @@
+> `/tmp/chip_tool_config.ini` file. Deleting this and other `.ini` files in the
+> `/tmp` directory can sometimes resolve issues related to stale configuration.
+
+<hr>
+
+<a name="building"></a>
+
+## Building and running the CHIP Tool
+
+Before you can use the CHIP Tool, you must compile it from source on Linux
+(amd64/aarch64) or macOS. If you want to run it on Raspberry Pi, it must use a
+64-bit OS.
+
+> **Note:** To ensure compatibility, always build the CHIP Tool and the Matter
+> device from the same revision of the `connectedhomeip` repository.
+
+### Building the CHIP Tool
+
+To build and run the CHIP Tool:
+
+1. Install all required packages for Matter and prepare the source code and the
+   build system. Read the [Building Matter](BUILDING.md) guide for instructions.
+2. Open a command prompt in the `connectedhomeip` directory.
+3. Run the following command:
+
+    ```
+    ./scripts/examples/gn_build_example.sh examples/chip-tool BUILD_PATH
+    ```
+
+    In this command, `BUILD_PATH` specifies where the target binaries are to be
+    placed.
+
+### Running the CHIP Tool
+
+To check if the CHIP Tool runs correctly, execute the following command from the
+`BUILD_PATH` directory:
+
+```
+$ ./chip-tool
+```
+
+As a result, the CHIP Tool prints all available commands. These are called
+_clusters_ in this context, but not all listed commands correspond to the
+_clusters_ in the Data Model (for example, pairing or discover commands). Each
+listed command can however become the root of the new more complex command by
+appending it with sub-commands. Examples of specific commands and their use
+cases are described in the [Supported commands and options](#commands) section.
+
+<hr>
+
+<a name="using"></a>
+
+## Using CHIP Tool for Matter device testing
+
+This section describes how to use CHIP Tool to test the Matter device. The
+following steps depend on the application clusters that you implemented on the
+device.
+
+This tutorial is using the
+[Matter lighting app example](https://github.com/project-chip/connectedhomeip/tree/master/examples/lighting-app)
+with the Bluetooth LE commissioning method support. You can use other Matter
+examples and still follow this procedure. If you use a different example, the
+[Step 7](#step-7-control-application-data-model-clusters) may vary depending on
+the clusters implemented in your application.
+
+### Step 1: Prepare the Matter device
+
+Build and program the device with the Matter device firmware by following the
+[Matter lighting-app example](https://github.com/project-chip/connectedhomeip/tree/master/examples/lighting-app)
+documentation.
+
+### Step 2: Enable Bluetooth LE advertising on Matter device
+
+Some examples are configured to advertise automatically on boot. Other examples
+require physical trigger, for example pushing a button. Follow the documentation
+of the Matter device example to learn how Bluetooth LE advertising is enabled
+for the given example.
```

Comment:
```suggestion
Some examples are configured to advertise automatically on boot. Other examples
require physical trigger, for example pushing a button.

For the Lighting Application Example, press **Button 4** to enable the Bluetooth LE advertising.
If you are using a different example, check its documentation
for information about how to enable the Bluetooth LE advertising.
```

Reply:
I think we cannot assume anything about how the BLE advertising is started, the fact that the button 4 is used is specific for our platform example, but the chip-tool-guide must remain generic.

Reply:
But we stopped being generic when we wrote that lighting app example is used for reference.

Reply:
Will be reformatted to:

"For the Lighting Application Example for the nRF Connect platform, press **Button 4** to enable the Bluetooth LE advertising."

Reply:
Lighting app doesn't mean nRF connect lighting-app :) Please don't do it. We used to have battles with other developers who complained that we're trying to make "general" guides tailored to our products.

Reply:
OMG :)

Ahh ok, I agreed because we are already linking the Matter Lighting Application Example, but it's:
https://github.com/project-chip/connectedhomeip/tree/master/examples/lighting-app

and not

https://github.com/project-chip/connectedhomeip/tree/master/examples/lighting-app/nrfconnect

That being said, we indeed must remove all nrfconnect occurrences.